packages/core-plugin-api/api-report.md

```diff
@@ -220,6 +220,8 @@ export type BackstagePlugin<
   getId(): string;
   getApis(): Iterable<AnyApiFactory>;
   getFeatureFlags(): Iterable<PluginFeatureFlagConfig>;
+  getInfo(): Promise<PluginInfo>;
+  setMetadataExtender(extender: ExtendMetadata): void;
```

Comment:
I'm thinking we aim to make this private API so that there's no need to consider the stability of it, what do you think? Most likely a hidden `__InternalSetMetadataExtender` is the way to go, possibly with some versioning information. My experience with using symbols or weak maps is that it just makes things more difficult to debug without much gained.